src/Servers/Kestrel/Core/src/Internal/Http/HttpResponseHeaders.cs

```diff
@@ -18,6 +19,13 @@ internal sealed partial class HttpResponseHeaders : HttpHeaders
         private static ReadOnlySpan<byte> CrLf => new[] { (byte)'\r', (byte)'\n' };
         private static ReadOnlySpan<byte> ColonSpace => new[] { (byte)':', (byte)' ' };
 
+        public Func<string, Encoding?> EncodingSelector { get; set; }
+
+        public HttpResponseHeaders(Func<string, Encoding?>? encodingSelector = null)
```

Comment:
Instead of nullable, why not set the default value to KestrelServerOptions.DefaultHeaderEncodingSelector? I think I see this in a few places.

Reply:
Nvm it's not possible